modules/dnn/src/layers/convolution_arm/fast_conv_2d.hpp

```diff
@@ -0,0 +1,699 @@
+        int ksize = Hk*Wk;
+        int padded_ksize = ((ksize + FX_VEC_NLANES-1)/FX_VEC_NLANES)*FX_VEC_NLANES;  // this code aims to let memory fit with vector size.
+        int nweights = Ick*padded_ksize;
+        conv->weightsPtr = (float*)malloc(nweights*sizeof(conv->weightsPtr[0]));
+        memset(conv->weightsPtr, 0, nweights*sizeof(conv->weightsPtr[0]));
+        for(int c = 0; c < Ick; c++) {
+            for (int k = 0; k < ksize; k++)
+                conv->weightsPtr[c*padded_ksize + k] = weightsPtr[c*ksize + k];
+        }
+    } else {
+        // the weights are packed as
+        // ngroups x (ceil((K/ngroups)/FX_CONV_MR)*FX_CONV_MR) x (Cg*Hk*Wk) x FX_CONV_MR tensor
+        int Kg = Ock/ngroups, Cg = max(Ick/ngroups, 1);
+        int Kg_aligned = ((Kg + FX_CONV_MR - 1)/FX_CONV_MR)*FX_CONV_MR;
+        size_t nweights = ngroups*Kg_aligned*Cg*Hk*Wk;
+        conv->weightsPtr = (float*)malloc(nweights*sizeof(conv->weightsPtr[0]));
+        memset(conv->weightsPtr, 0, nweights*sizeof(conv->weightsPtr[0]));
+        float* packed_wptr = conv->weightsPtr;
+
+        // pack the weight.
+        for(int g = 0; g < ngroups; g++) {
+            for(int k0 = 0; k0 < Kg_aligned; k0 += FX_CONV_MR) {
+                int dk = Kg - k0 < FX_CONV_MR ? Kg - k0 : FX_CONV_MR;
+                for(int c = 0; c < Cg; c++) {
+                    for(int yx = 0; yx < Hk*Wk; yx++, packed_wptr += FX_CONV_MR) {
+                        const float* wptr = weightsPtr + ((g*Kg + k0)*Cg + c)*Hk*Wk + yx;
+                        int k = 0;
+                        for(; k < dk; k++, wptr += Cg*Hk*Wk)  // wptr change the line every time.
+                            packed_wptr[k] = *wptr;
+                        for(; k < FX_CONV_MR; k++)
+                            packed_wptr[k] = 0.f;
+                    }
+                }
+            }
+        }
+
+    }
+    return conv;
+}
+
+void fast_depthwise_conv2d(InputArray _input, OutputArray _output,
+                           const struct fast_conv2d_arm* conv)
+{
+    Mat input = _input.getMat();
+    Mat output = _output.getMat();
+
+    MatShape inpsize = shape(input);
+    MatShape outsize = shape(output);
+
+    CV_Assert(inpsize.size() == 4 && inpsize[0] == outsize[0] && outsize[1] == conv->Ock && inpsize[1] == conv->Ick);
+    CV_Assert(conv->ngroups > 1 && conv->ngroups == conv->Ock && conv->Ock == conv->Ick);
+    int N = inpsize[0], C = inpsize[1], Hi = inpsize[2], Wi = inpsize[3];
+    int Hk = conv->Hk, Wk = conv->Wk;
+    int H0 = outsize[2], W0 = outsize[3];
+    const size_t inp_planesize = (size_t)Hi*Wi;
+    const size_t out_planesize = (size_t)H0*W0;
+    float minval = conv->minval, maxval = conv->maxval;
+    bool ifMinMaxAct = conv->ifMinMaxAct;
+    const ActivationLayer* activ = conv->activ;
+
+    int stride_y = conv->stride_y, stride_x = conv->stride_x;
+    int dilation_y = conv->dilation_y, dilation_x = conv->dilation_x;
+
+//    int pad_top = 0, pad_bottom = 0;
```

Comment:
you removed padding and instead add padding before running convolution. This is very bad solution in terms of performance. Please, put it back